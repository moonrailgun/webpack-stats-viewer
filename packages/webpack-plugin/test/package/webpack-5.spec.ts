// const webpack5 = require('webpack');
// const MemoryFS = require('memory-fs');
// const { advanceTo } = require('jest-date-mock');
// const { set } = require('lodash');
import webpack5 from 'webpack';
import MemoryFS from 'memory-fs';
import {set} from 'lodash';
import {advanceTo} from 'jest-date-mock'

const config = require('./app/webpack.config.js')

advanceTo(new Date(2020, 10, 30));

jest.setTimeout(10 * 1000);

describe('webpack plugin package', () => {
  test('webpack5', (done) => {
    expect.assertions(3);

    const compiler = webpack5(config);
    compiler.outputFileSystem = new MemoryFS();

    compiler.run((error, stats) => {
      expect(error).toEqual(null);
      if(!stats) {
        throw new Error();
      }
      expect(stats.hasErrors()).toBe(false);

      // Set bundle-stats.html size to 0 to ignore size changes
      const { assets } = stats.toJson({ source: false, assets: true });
      if(!assets) {
        throw new Error();
      }
      set(assets, '[0].size', 0);
      set(assets, '[0].info.size', 0);

      expect(assets).toMatchSnapshot();
      done();
    });
  });
});
