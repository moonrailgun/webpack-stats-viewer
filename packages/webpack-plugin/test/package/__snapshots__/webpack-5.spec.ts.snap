// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack plugin package webpack5 1`] = `
[
  {
    "auxiliaryChunkIdHints": [],
    "auxiliaryChunkNames": [],
    "auxiliaryChunks": [],
    "cached": false,
    "chunkIdHints": [],
    "chunkNames": [
      "main",
    ],
    "chunks": [
      179,
    ],
    "comparedForEmit": false,
    "emitted": true,
    "filteredRelated": undefined,
    "info": {
      "javascriptModule": false,
      "minimized": true,
      "size": 0,
    },
    "isOverSizeLimit": false,
    "name": "main.js",
    "related": {},
    "size": 0,
    "type": "asset",
  },
]
`;
